aten/src/ATen/native/native_functions.yaml

```diff
@@ -83,18 +83,22 @@
 - func: dropout(Tensor input, float p, bool train) -> Tensor
 
 - func: dropout_(Tensor(a!) self, float p, bool train) -> Tensor(a!)
+  variant: function, method
```

Comment:
I'd prefer to have an indication in native_functions that something is wrong; it seems totally feasible that someone would copy-paste one of these, not happen to call the function variant and not get a warning.

I'm not sure if this would work because not all of these are deprecated, but I'd consider:
1) adding an annotation, e.g. "deprecated_method: True"
2) If you have one of these bad methods that don't have a deprecated_method tag (and isn't in NN?) you error out.

Is that feasible?

Reply:
Do you think it would be simpler if we report an error in `gen_python_functions.py` (where TORCH_WARN is suggested), if the function is not one of the currently existing ones? I don't think anyone should add a new deprecated inplace operator that can be used as a function at this point (we are kind of inconsistent about whether an inplace operator has a function variant at this point).

And the time window to add a new such operator is probably short because we can actually come back and revisit this issue in two weeks, assuming a new release comes out before the end of July :)

Reply:
a whitelist also seems fine, as long as we still TORCH_WARN for those.